### PR TITLE
Trim Document Key

### LINKED ISSUE #5317: Route not Found exception for documents ending with space.
- Create a new document ending with space - "test ".
- Open document.

error

route error

### PR DESCRIPTION
**Pull request by neha04**
## Changes in this pull request 
Resolves #5317

### CODE REVIEW

**Comment by BlackbitNeueMedien** [comment]
I also stumbled upon this for object keys. These are also not trimmed. Is this on purpose, @weisswurstkanone / @brusch ?

**Comment by brusch** [comment]
@BlackbitNeueMedien No, i guess it would be good to trim them as well, I can't think of a use-case where it wouldn't make sense. 

@neha04 can you update your PR please? Thanks a lot! :)

File: models/Element/Service.php [hunk 1]
```diff
@@ -940,6 +940,8 @@ public static function getValidKey($key, $type)
         if ($type === 'object') {
             $key = preg_replace('/[<>]/', '-', $key);
         } elseif ($type === 'document') {
+            $key = trim($key);
```
Comment (dvesh3) [inline models/Element/Service.php]: just move this before ifelse should be enough and remove trim for else part.

**Comment by brusch** [comment]
Nice, thanks a lot!

**Pull request merged**